packages/path-compare/src/index.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Comparison function for two values of the same type `T`.
+ *
+ * Can be used with `Array.prototype.sort` and other similar contexts
+ *
+ * @template T The type of the values to compare
+ * @callback CompareFn
+ * @param {T} a First value
+ * @param {T} b Second value
+ * @returns {number} Negative integer if `a < b`; positive integer if `a > b`;
+ * `0` or `NaN` if equal
```

Comment:
```suggestion
 * `0` if equal
```
(other packages in this project interpret `NaN` to mean "incomparable")